SparkPost/matchbox
# UX-769 Add Progress Component

### What Changed
- Adds new Progress component

### How To Test or Verify
- Run Libby and visit: http://localhost:9001/?path=Progress__basic-usage
- Verify the component works correctly and matches the mocks here:
  - https://www.figma.com/file/uKubC4RT18p0MZnGSJI1KA/Progress?node-id=4%3A38

### PR Checklist
